lib/smart_proxy_dns_powerdns/dns_powerdns_main.rb

```diff
@@ -83,8 +83,8 @@ def remove_ptr_record(ptr)
 
     def do_remove(name, type)
       zone = get_zone(name)
-      if delete_record(zone['id'], name, type)
-        raise Proxy::Dns::Error.new("Failed to remove record #{name} #{type}") unless rectify_zone(zone['name'])
+      unless delete_record(zone['id'], name, type) and update_soa(zone['id'], zone['name']) and rectify_zone(zone['name'])
```

Comment:
You are right, I totally overlooked that one! I will change it back to the way it was before.